src/Database/Schema/SqliteSchema.php

```diff
@@ -355,6 +355,100 @@ public function constraintSql(Table $table, $name)
     /**
      * {@inheritDoc}
      */
+    public function addConstraintSql(Table $table)
+    {
+        $tmpTableName = $this->_driver->quoteIdentifier('tmp_' . $table->name());
+        $tableName = $this->_driver->quoteIdentifier($table->name());
+
+        $columns = $indexes = $constraints = [];
+        foreach ($table->columns() as $column) {
+            $columns[] = $this->columnSql($table, $column);
+        }
+
+        foreach ($table->constraints() as $constraint) {
+            $constraints[] = $this->constraintSql($table, $constraint);
+        }
+
+        foreach ($table->indexes() as $index) {
+            $indexes[] = $this->indexSql($table, $index);
+        }
+        $createSql = $this->createTableSql($table, $columns, $constraints, $indexes);
+
+        $columnsList = implode(', ', $table->columns());
+        $copySql = sprintf(
+            'INSERT INTO %s(%s) SELECT %s FROM %s',
+            $tableName,
+            $columnsList,
+            $columnsList,
+            $tmpTableName
+        );
+
+        $dropSql = sprintf('DROP TABLE IF EXISTS %s', $tmpTableName);
+
+        $sql = [
+            $dropSql,
+            sprintf('ALTER TABLE %s RENAME TO %s', $tableName, $tmpTableName),
+            $createSql[0],
+            $copySql,
+            $dropSql
+        ];
+
+        return $sql;
```

Comment:
Tell me about it... FYI, there is a drop at the beginning and at the end because in the tests, I sometimes get a weird race condition (or that is was I suspect it is) where the table is locked and the last drop can't be performed because it is still busy doing the data copy. Which leads to a big mess because the next iteration of dealing with constraints raises errors (table can't be renamed because a table with the same name already exists and so on).
The only workaround I could find was to add a DROP as the first query to clear any leftovers.

I did some search and the proposed solution is to use ``closeCursor()`` when executing multiple queries in a row with SQLite, which is done, so I don't really understand that error.

And it gets weirder... I get that error from a specific test ([this one](https://github.com/cakephp/cakephp/pull/7503/files#diff-c22681210f8a3908b6e2b9cb94148ba1R482)) which is no different from the test right underneath which pass with no problem. Full error (there is a debug still lying around) can be found on [this build](https://travis-ci.org/cakephp/cakephp/jobs/83401317).

So if you guys have any idea...